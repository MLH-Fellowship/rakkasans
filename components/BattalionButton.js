import React from "react";
import { View, Image, Text, StyleSheet, TouchableOpacity } from "react-native";

import Colors from "../constants/Colors";
import Dimensions from "../constants/Dimensions";

export default function BattalionButton({
  onPress,
  image,
  iconWidth,
  iconHeight,
  text,
}) {
  return (
    <View style={styles.container}>
      <TouchableOpacity
        style={[styles.button, Dimensions.button]}
        onPress={onPress}
      >
        <Image
          source={image}
          style={{ width: iconWidth, height: iconHeight }}
        />
      </TouchableOpacity>
      <Text style={[styles.text, Dimensions.font]}>{text}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  button: {
    padding: 1,
    height: 85,
    width: 85,
    alignItems: "center",
    justifyContent: "center",
  },
  text: {
    textAlign: "center",
    color: Colors.darkGray,
    paddingTop: 8,
  },
  container: {
    alignItems: "center",
    justifyContent: "center",
  },
});
