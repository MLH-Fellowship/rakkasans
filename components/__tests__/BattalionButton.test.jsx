import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { Image } from 'react-native';
import renderer from 'react-test-renderer';

import BattalionButton from '../BattalionButton';

import Batt133Pic from '../../assets/images/battalions/1-33.png';

jest.useFakeTimers();
configure({ adapter: new Adapter() });

describe('BattalionButton', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<BattalionButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with an image', () => {
    const wrapper = shallow(
      <BattalionButton>
        <Image source={Batt133Pic} />
      </BattalionButton>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('handles onPress', () => {
    const onPress = jest.fn();

    const wrapper = shallow(
      <BattalionButton onPress={onPress}>
        <Image source={Batt133Pic} />
      </BattalionButton>,
    );

    wrapper
      .find('TouchableOpacity')
      .first()
      .props()
      .onPress();

    expect(onPress).toHaveBeenCalledTimes(1);
  });
});
