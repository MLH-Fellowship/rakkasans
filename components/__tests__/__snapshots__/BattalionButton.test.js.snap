// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BattalionButton renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 15,
        "elevation": 5,
        "height": 85,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 1,
        "shadowColor": "#a0a0a0",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.4,
        "shadowRadius": 8,
        "width": 85,
      }
    }
  >
    <Image
      style={
        Object {
          "height": undefined,
          "width": undefined,
        }
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#333333",
          "paddingTop": 8,
          "textAlign": "center",
        },
        Object {
          "fontFamily": "andale-mono",
        },
      ]
    }
  />
</View>
`;

exports[`BattalionButton renders correctly with an image 1`] = `ShallowWrapper {}`;
